### PR TITLE
Bug fixes after pubs run

### PR DESCRIPTION
**Pull request by AnishReddyRavula**
- No changes to the chameleon used in the research view, but it is showing up in changes

- Fix in data migration to have the appropriate flag for `found_by_algo`
- Only approved_with 'Publication' will be `cited_by_chameleon`
- Renamed `reviewed` to `checked_for_duplicates`
- renamed `flag_duplicates` to `get_duplicate_pubs` so the caller can handle the duplicates
- No changes to the chameleon used in the research view, but it is showing up in changes
- In deduplicates - pick the right duplicate method ( flag 'preprint' as duplicate)
- Made the reviewing publications function more robust

Not sure why so many changes. But it is what it is

### CODE REVIEW

File: projects/user_publication/gscholar.py
Comment (super-cooper) [inline projects/user_publication/gscholar.py:207]: `Publication.year` is a non-null integer field. I would be very surprised if we found a publication without a year, but also, if there _isn't_ a valid year, saving this `Publication` will fail.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:37]: This check is unnecessary. A loop through an empty `QuerySet` is a no-op.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:40]: This `continue` is unnecessary as well. This is at the end of the loop anyway.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:140]: This should not be set if the publication is rejected.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:150]: I'm not sure what this function is for. These publications are usually reviewed in the admin interface.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:214]: This call will fail if the filter is empty

File: projects/user_publication/deduplicate.py
Comment (AnishReddyRavula) [reply projects/user_publication/deduplicate.py:35]: My bad. I did not frame the doc right. If the passed pubs are not stored (saved) in the database the function fails with an error. Will correct it

File: projects/user_publication/gscholar.py
Comment (AnishReddyRavula) [reply projects/user_publication/gscholar.py:207]: Yes, it will skip the publication. As for Gscholar, all we are doing is scraping. It sometimes may not have a year to the publication. In my runs, there is one publication for which the whole script will fail, and it's really annoying. So I included the catch here. Will log it and skip the publication.

File: projects/user_publication/publications.py
Comment (AnishReddyRavula) [reply projects/user_publication/publications.py:37]: I would like to skip the publication if there is at least a single publication with the same attributes after updating all the same publications so need that check.

File: projects/user_publication/publications.py
Comment (AnishReddyRavula) [reply projects/user_publication/publications.py:140]: yeah. It is confusing. The meaning of this is, it was tried for approval with email and then got rejected from email.
This helps us maintain a count of how many got rejected from email.

File: projects/user_publication/publications.py
Comment (AnishReddyRavula) [reply projects/user_publication/publications.py:150]: I think the database I had a few unreviewed publications so I ended up implementing this. After this change, I don't think doing this admin interface is going to be easy. I will have to make changes admin interface to support this. Will create a Trello card for this.

Comment (AnishReddyRavula) [reply projects/user_publication/publications.py:150]: I am removing this and also updated the admin interface to display source information allowing to make approvals on the publications page itself.

File: projects/user_publication/deduplicate.py
Comment (super-cooper) [inline projects/user_publication/deduplicate.py]: I'm confused by this. Both of these branches have the same body. Could we not just check 
```python
if duplicate_has_preprint:
    duplicate_pub = dpub
    original_pub = opub
else:
    duplicate_pub = opub
    original_pub = dpub
```

File: projects/user_publication/publications.py
Comment (super-cooper) [reply projects/user_publication/publications.py:140]: I don't think we should be using the `approved_with` field to describe how a publication is rejected. I think this should be left null. 

If a publication is rejected, it's basically for two reasons:

1. It's very obvious (published in 1999 or something like that)
2. I emailed the PI and they said they didn't use Chameleon

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:130]: My comments below aside, what is this function for? As mentioned in the last review, publication reviews are usually done in the admin interface. Email reviews usually take time, and sometimes never get responses at all, so having a CLI for this seems cumbersome. Much easier to send the review and then click one button in the admin interface.

File: projects/user_publication/publications.py
Comment (super-cooper) [inline projects/user_publication/publications.py:109]: Come to think of it, what is this status for? If the publication hasn't been reviewed, then it hasn't been approved. So how is something approved with a pending review?

File: projects/user_publication/publications.py
Comment (super-cooper) [reply projects/user_publication/publications.py:195]: My bad 😅

File: projects/user_publication/publications.py
Comment (AnishReddyRavula) [reply projects/user_publication/publications.py:109]: This is just to indicate that the publication source still needs approval.
I think it is better to remove this and add another field to indicate the source needs a review.

**Review by super-cooper — APPROVED** [review]
Nicely done on the new changes :D

**Pull request merged**
